source/inputoutput.rst

```diff
@@ -0,0 +1,116 @@
+Inputs and Outputs in BharatSim
+===============================
+
+Inputs
+------
+**to be completed**
+
+* function should feed in the csv file 1 row at a time
+* data gets 'ingested' and graphNodes are created
+
+Outputs
+-------
+
+A convenient way to store the output is by using a `CSV <https://en.wikipedia.org/wiki/Comma-separated_values>`_ file, which is discussed below.
+
+.. note:: In case the quantities you'd like to output are fairly simple, you could use Scala's ``println`` function to directly output what you need.
+
+Outputting to a CSV file
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The BharatSim engine contains a class called ``CsvOutputGenerator``, which has two attributes:
+* ``path``, the desired path for the output file to be stored
+* ``csvSpecs``, a user-defined class that outputs the headers and the rows required. Note that this user-defined class should extend the ``CSVSpecs`` trait and override the ``getHeaders`` and ``getRows`` methods.
+
+Once the class is complete, we now have to **register** it in the simulation (defined in the main function), similar to how we registered `agents <#>`_. First, we must import ``CsvOutputGenerator`` and ``SimulationListenerRegistry``
+
+.. code-block:: scala
+   
+  import com.bharatsim.engine.listeners.{CsvOutputGenerator, SimulationListenerRegistry}
+
+
+Next, we register it using the ``register`` method of ``SimulationListenerRegistry``
+
+.. code-block:: scala
+
+  SimulationListenerRegistry.register(
+    new CsvOutputGenerator("src/main/resources/output.csv", new myCsvSpecs(context))
+    )
+
+where ``myCsvSpecs`` is the user-defined class which requires the context as an attribute.
+
+An example: The SIR Model
```

Comment:
After some careful consideration, I have decided that this entire example should actually be part of the "Writing your first program" section. That way, the context required for the code at the end will follow naturally. 

As a result, you need a new example here. Please try to come up with one. It should be very simple, and should not involve the more complicated aspects of the code like ``infectionStatus`` and so on if possible. 

As I've mentioned above, let's do it _without_ the Listener Registry (provided I'm right and this is possible). It's ok if it's trivial, like fetching all the ``Home``s and printing out their latitude and longitude, for example. This should only use other methods defined in the "Framework Basics" section. Go through the ``GISOutputSpec.scala`` class in the ``model`` package to get a sense of how to print multiple rows to a CSV file in one go.